src/open_company/api/entry.clj

```diff
@@ -0,0 +1,50 @@
+(ns open-company.api.entry
+  (:require [compojure.core :refer (defroutes GET OPTIONS)]
+            [liberator.core :refer (defresource by-method)]
+            [open-company.api.common :as common]
+            [open-company.resources.company :as company-res]
+            [open-company.representations.common :as common-rep]
+            [open-company.representations.company :as company-rep]
+            [cheshire.core :as json]))
+
+;; ----- Representations
+;; I assume these could go into a separate representations ns but since
+;; they're just links and not full domain objects I just kept them here
+
+(defn company-link [{:keys [name] :as c}]
+  (common-rep/link-map "company" "GET" (company-rep/url c) company-rep/media-type :name name))
+
+(defn links [user companies]
+  (cond-> [(common-rep/link-map "company-list" "GET" "/companies/" company-rep/collection-media-type)]
+    user      (conj (common-rep/link-map "company-create" "POST" "/companies/" company-rep/media-type))
+    companies (into (mapv company-link companies))))
+
+;; ----- Resources - see: http://clojure-liberator.github.io/liberator/assets/img/decision-graph.svg
+
+(defresource entry-point []
+  common/anonymous-resource
+
+  :allowed-methods [:options :get]
+  ;; slug supplied to allow-anonymous here is irrelevant.
+  ;; the allowed? checking and token decoding should be
+  ;; decoupled and probably use :initialize-context for
+  ;; token decoding
+  :allowed? (fn [ctx] (common/allow-anonymous "foo" ctx))
```

Comment:
Totally agree. I thought this would better be done in another PR after this is merged as it affects all API namespaces and would make things slightly messy to review if done in this one as well.